src/charm.py

```diff
@@ -0,0 +1,415 @@
+            return False
+        logger.info("Config file is written")
+        return True
+
+    @property
+    def _pfcp_agent_config_file_is_written(self) -> bool:
+        """Returns whether the pfcp agent config file was written to the workload container.
+
+        Returns:
+            bool: Whether the pfcp agent config file was written
+        """
+        if not self._pfcp_agent_container.exists(
+            f"{PFCP_AGENT_CONTAINER_CONFIG_PATH}/{CONFIG_FILE_NAME}"
+        ):
+            logger.info(f"Config file is not written: {CONFIG_FILE_NAME}")
+            return False
+        logger.info("Config file is written")
+        return True
+
+    def _on_bessd_pebble_ready(self, event: PebbleReadyEvent) -> None:
+        """Handle Pebble ready event for bessd container.
+
+        Args:
+            event: PebbleReadyEvent
+        """
+        if not self._bessd_config_file_is_written:
+            self._write_bessd_config_file()
+        if not self.access_network_gateway:
+            self.unit.status = BlockedStatus("Please set a value for the `access-gateway` config.")
+            return
+        if not self.core_network_gateway:
+            self.unit.status = BlockedStatus("Please set a value for the `core-gateway` config.")
+            return
+        if not self.gnb_subnet:
+            self.unit.status = BlockedStatus("Please set a value for the `gnb-subnet` config.")
+            return
+        try:
+            self._exec_command_in_bessd_workload(
+                command=[
+                    "ip",
+                    "route",
+                    "replace",
+                    self.gnb_subnet,
+                    "via",
+                    self.access_network_gateway,
+                ]
+            )
+            self._exec_command_in_bessd_workload(
+                command=[
+                    "ip",
+                    "route",
+                    "replace",
+                    "default",
+                    "via",
+                    self.core_network_gateway,
+                    "metric",
+                    "110",
+                ],
+            )
+            self._exec_command_in_bessd_workload(
+                command=[
+                    "iptables",
+                    "-I",
+                    "OUTPUT",
+                    "-p",
+                    "icmp",
+                    "--icmp-type",
+                    "port-unreachable",
+                    "-j",
+                    "DROP",
+                ],
+            )
+        except ExecError:
+            self.unit.status = WaitingStatus("Cannot execute command in workload, waiting.")
+            event.defer()
+            return
+        self._bessd_container.add_layer("upf", self._bessd_pebble_layer, combine=True)
+        self._bessd_container.replan()
+        self._exec_command_in_bessd_workload(
+            command=["bessctl run /opt/bess/bessctl/conf/up4"],
```

Comment:
```suggestion
            command=["bessctl run /opt/bess/bessctl/conf/up4".split()],
```
This will separate the tokens and prevent weird shenanigans with quoting

Reply:
Fixed by using commas everywhere